### PR TITLE
Feat/create poll

### PR DESCRIPTION
**Pull request by Leon-pix-coder**
Create poll UI Components

### CODE REVIEW

File: src/components/ButtonRadio/index.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import clsx from "clsx";
+import { useEffect, useState } from "react";
+
+type IButtonRadioOption = {
```
Comment (evan-spidy) [inline src/components/ButtonRadio/index.tsx]: if it is a type then you do not prefix with I

File: src/components/Drawer/index.tsx [hunk 2]
```diff
@@ -7,7 +7,7 @@ interface IDrawerProps {
   direction?: "left" | "right" | "top" | "bottom";
   children: ReactNode | ReactNode[];
   rootClassName?: string;
-  onClose?: () => void;
+  onClose?: (v: boolean) => void;
```
Comment (evan-spidy) [inline src/components/Drawer/index.tsx]: what is v? give a better name

File: src/components/InputGroup/index.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import React, { useState } from "react";
+import Input from "../Input";
+
+interface Option {
```
Comment (evan-spidy) [inline src/components/InputGroup/index.tsx]: this should be type

File: src/components/SimpleDatePicker/index.tsx [hunk 4]
```diff
@@ -2,49 +2,93 @@ import { DayPicker, WeekdayProps } from "react-day-picker";
...
+        isVisible={isVisible}
+        direction="bottom"
+        onClose={setIsVisible}
+        rootClassName="px-[17.5px] pt-5 pb-7 bg-tertiary"
       >
-        Confirm
-      </button>
-    </Drawer>
+        <DayPicker
+          {...dayPickerProps}
```
Comment (evan-spidy) [inline src/components/SimpleDatePicker/index.tsx]: spread operator should always put it at the end so developer can pass in and overwrite default value

File: src/components/ToggleSlider/index.tsx [hunk 5]
```diff
@@ -18,6 +19,10 @@ const ToggleSlider = ({ items = [] }: IToggleSlider) => {
     setActiveIndex(index);
   };
 
+  useEffect(() => {
+    onChange?.(activeIndex)
+  }, [activeIndex, onChange])
+
```
Comment (evan-spidy) [inline src/components/ToggleSlider/index.tsx]: why do you need a useEffect here? you can move the line into line 19

File: src/components/Upload/index.tsx [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+  };
+
+  const handleUpload = async (imageFile: Blob) => {
+    try {
+      if (!imageFile) return;
+      const data = new FormData();
+      data.append("file", imageFile, 'image/png');
+      data.append("chainId", chainId);
+      const result = await uploadWithToken("/api/app/file/upload", data);
+      console.log(result);
```
Comment (evan-spidy) [inline src/components/Upload/index.tsx]: why do you need to log here?

File: src/pageComponents/Home/index.tsx [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+        htmlElement.style.cssText = styles;
+      }
+    }
+  }, []);
+
+  const fetchForYouData = async (alias: string[] = []) => {
+    const { data } = await postWithToken("/api/app/discover/random-app-list", {
+      chainId,
+      alias,
+      category: "ForYou",
```
Comment (evan-spidy) [inline src/pageComponents/Home/index.tsx]: check if this is in constants file

File: src/pageComponents/Home/index.tsx [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+    if (alias.length > 0) {
+      currentForyouPage.current++;
+    }
+  };
+
+  const fetchRecommendData = async () => {
+    const { data } = await postWithToken("/api/app/discover/random-app-list", {
+      chainId,
+      category: "Recommend",
```
Comment (evan-spidy) [inline src/pageComponents/Home/index.tsx]: same for this

File: src/utils/canvasUtils.ts [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+  });
+
+export function getRadianAngle(degreeValue: number): number {
+  return (degreeValue * Math.PI) / 180;
+}
+
+interface Size {
+  width: number;
+  height: number;
+}
```
Comment (evan-spidy) [inline src/utils/canvasUtils.ts]: move type into another file

File: src/components/ButtonRadio/index.tsx (hunk 1 above)
Comment (Leon-pix-coder) [reply src/components/ButtonRadio/index.tsx]: Updated

File: src/components/Drawer/index.tsx (hunk 2 above)
Comment (Leon-pix-coder) [reply src/components/Drawer/index.tsx]: Updated

File: src/components/InputGroup/index.tsx (hunk 3 above)
Comment (Leon-pix-coder) [reply src/components/InputGroup/index.tsx]: Updated

File: src/components/SimpleDatePicker/index.tsx (hunk 4 above)
Comment (Leon-pix-coder) [reply src/components/SimpleDatePicker/index.tsx]: Updated

File: src/components/Upload/index.tsx (hunk 6 above)
Comment (Leon-pix-coder) [reply src/components/Upload/index.tsx]: Updated

File: src/pageComponents/Home/index.tsx (hunk 7 above)
Comment (Leon-pix-coder) [reply src/pageComponents/Home/index.tsx]: Updated

File: src/pageComponents/Home/index.tsx (hunk 8 above)
Comment (Leon-pix-coder) [reply src/pageComponents/Home/index.tsx]: Updated

File: src/utils/canvasUtils.ts (hunk 9 above)
Comment (Leon-pix-coder) [reply src/utils/canvasUtils.ts]: Updated

**Pull request merged**